x/auth/ante/unordered.go

```diff
@@ -62,7 +78,39 @@ func (d *UnorderedTxDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, _ bool, ne
 		return ctx, errorsmod.Wrapf(sdkerrors.ErrInvalidRequest, "unordered tx ttl exceeds %d", d.maxUnOrderedTTL)
 	}
 
-	txHash := sha256.Sum256(ctx.TxBytes())
+	// in order to create a deterministic hash based on the tx, we need to hash the contents of the tx with signature
```

Comment:
Would love if we separated the hashing logic part from the unordered tx ante; eg put ths into a separate hash function so it can be better audited and changed

Reply:
done,